### PR TITLE
fixed

### PR DESCRIPTION
**Pull request by LizaBalshv**

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 1]
```diff
@@ -3,17 +3,48 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+
+    private static final int MAX_ITEMS_NUMBER = 10;
+    private Object[] keys;
+    private Object[] values;
```
Comment (okuzan) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:9]:
```suggestion
    private K[] keys;
    private V[] values;
```
that's where we should use type parameters

**Review by okuzan — CHANGES_REQUESTED** [review]
Good job

**Review by okuzan — CHANGES_REQUESTED** [review]
for sync, ignore this comment

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,17 +3,45 @@
...
+
     @Override
     public void put(K key, V value) {
+        if (size < MAX_ITEMS_NUMBER) {
+            for (int i = 0; i < size; i++) {
+                if (key == null && keys[i] == null || key != null && key.equals(keys[i])) {
+                    values[i] = value;
+                    return;
+                }
+            }
```
Comment (okuzan) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:24]: image
repeated logic, please extract to a private method
you can return an index of an equal key, -1 if not found

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,17 +3,53 @@
...
+            int index = findKeyIndex(key);
+            if (index != -1) {
+                values[index] = value;
+            } else {
+                keys[size] = key;
+                values[size] = value;
+                size++;
+            }
+        } else {
+            System.out.println("The storage is full, it is impossible to make a new entry");
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:28]: For the future, it is not good practice to log data using such a way,
It is better to throw an Exception or use the logger
```suggestion
```

**Review by Ivan95kos — APPROVED** [review]
Great job with a minor comment

**Pull request closed without merging**